pkg/agent/utils.go

```diff
@@ -276,7 +283,11 @@ func GetOrderedKubeletConfigFlagString(k map[string]string, cs *datamodel.Contai
 	keys := []string{}
 	for key := range k {
 		if !kubeletConfigFileEnabled || !TranslatedKubeletConfigFlags[key] {
-			keys = append(keys, key)
+			// --node-status-report-frequency is not a valid command line flag
+			// and should only be explicitly set in the config file
+			if key != nodeStatusReportFrequencyFlag {
```

Comment:
how about creating a new map of "this is not in config file"?

would be nice to avoid hardcoding single key names here